Fix wildcard field name example by introducing ** (#1337)
brimsec/zq: zql/docs/search-syntax/README.md

### Bare Word

> The simplest form of such a search is a "bare" word (not wrapped in quotes), which will match against any field that contains the word, whether it's an exact match to the data type and value of a field or the word appears as a substring in a field.

Author: @henridf: I made this change after seeing your PR review comment. Please give it a look.

Reviewer: 👍

### Role of Data Types

When working with named fields, the data type of the field becomes significant in two ways.

1. To match successfully, the value entered must be comparable to the data type of the named field. For instance, the `host` field of the `http` events in our sample data are of `string` type, since it logs an HTTP header that is often a hostname or an IP address.

   ```zq-command
   zq -t 'count() by host | sort count,host' http.log.gz
   ```

   #### Output:
   ```zq-output head:4
   #0:record[host:bstring,count:uint64]
   0:[0988253c66242502070643933dd49e88.clo.footprintdns.com;1;]
   0:[10.47.21.1;1;]
   0:[10.47.21.80/..;1;]
   ...
   ```

>    An attempted field/value match `host=10.47.21.1` would not match the event counted in the middle row of this table, since ZQL recognizes the bare value `10.47.21.1` as an IP address before comparing it to all the fields named `host` that it sees in the input stream. However, `host="10.47.21.1"` would match, since the quotes cause ZQL to treat the value as a string.

Author: @henridf: I made this change after seeing your PR review comment. Please give it another look.

Reviewer: Reading this I'm thinking of `literal`  in place of `bare value`. But your call, I'm guessing you might find that it's too progamming language-y for the goal of this doc.

Author: Yes, I'm inclined to leave it as-is. I'm realizing that at some point we'll surely write a "Zesty Guide For Developers" where these same concepts are explained in programmer-friendly terms that use different terminology and use "foo/bar" examples rather than detailed Zeek ones that would make more sense to the security users. For now, I've got buy-in from the team that with our limited docs cycles we'll continue to make things as accessible as we can to security users who may not have a programming background. Thanks for helping me think it through.

### Wildcard Field Names

> Since the data type of the value is considered in field/value matches, it's possible to search for the value across any fields of the value's type by entering a wildcard in place of the field name. Two wildcard operators are available depending on how broad you want your search to be. The `*` operator matches all top-level fields of the value's type, and the `**` operator additionally matches such values when they appear nested within records.

Reviewer: ```suggestion
It's possible to search for the typed value across _any_ field by entering a wildcard in place of the field name. Two wildcard operators are available depending on how broad you want your search to be. The `*` operator matches all top-level fields of the value's type, and the `**` operator additionally matches such values when they appear nested within records.
```

Reviewer 2: s/across any field/across all fields/

Author: @henridf @nwt: I've incorporated your feedback, but made a small additional adjustment. I respect that the leading "Since the data type..." text was arguably redundant, so I understand the instinct to remove it. At the same time (and maybe I'm not giving our readers enough credit here) I'm concerned that the shorthand "typed value" could be misinterpreted as "the value I typed into the Search box in Brim" as opposed to "the value of the data type that the ZQL parser interpreted the value to be". I don't expect confusion from users who do programming, but I also expect some non-programmer users coming to Brim/ZQL after using incumbent search tools may need more of a helping hand. Anyway, please do give it another look and let me know what you think.

> For example, the following search matches many `ssl` and `conn` events that contain the value `10.150.0.85` in `addr`-type fields of the `id` record, such as `id.resp_h`. It also matches `notice` events where it appears in `id.resp_h` and also `dst`, a top-level field also of the `addr` type. Compare this with our [bare word](#bare-word) example where we also matched as a substring of the `string`-type field named `certificate.subject`.

Reviewer: ```suggestion
For example, the following search matches many `ssl` and `conn` events that contain the value `10.150.0.85` in `addr`-type fields of the `id` record, such as `id.resp_h`. It also matches `notice` events where it appears in `id.resp_h` and also `dst`, a top-level field also of the `addr` type. Compare this with our [bare word](#bare-word) example where we also matched as a substring of the `string`-type field named `certificate.subject`. This highlights how search matches both on typed values and their string representation, whereas field/value match is stricter, and considers typed values only.
```

I think making the difference explicit is helpful (but perhaps with improved phrasing).
